tests/components/mqtt/conftest.py

```diff
@@ -49,6 +50,28 @@ def mock_temp_dir(temp_dir_prefix: str) -> Generator[str]:
         yield mocked_temp_dir
 
 
+@pytest.fixture
+def mock_debouncer(hass: HomeAssistant) -> Generator[asyncio.Event]:
+    """Mock EnsureJobAfterCooldown.
+
+    Returns an asyncio.Event that allows to await the debouncer task to be finished.
+    """
+    task_done = asyncio.Event()
+
+    class MockDeboncer(EnsureJobAfterCooldown):
+        """Mock the MQTT client (un)subscribe debouncer."""
+
+        async def _async_job(self) -> None:
+            """Execute after a cooldown period."""
+            await super()._async_job()
+            task_done.set()
+
+    with patch(
+        "homeassistant.components.mqtt.client.EnsureJobAfterCooldown", MockDeboncer
```

Comment:
I think this is great.  Normally we avoid patching the integration code, but in this case its OK as we generally have exceptions for sleeps/waits/debouncers/timers causing long run times in tests.

Maybe move the `EnsureJobAfterCooldown` to `homeassistant.components.mqtt.utils` so its clear its a more general utility and add a comment about why we are patching it directly

Reply:
https://github.com/home-assistant/core/pull/120392